queries.sql

```diff
@@ -7,4 +7,42 @@ SELECT name, escape_attempts from animals WHERE weight_kg >10.5;
 SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name='Pikachu';
 SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
-SELECT * FROM animals WHERE name NOT IN('Gabumon');
\ No newline at end of file
+SELECT * FROM animals WHERE name NOT IN('Gabumon');
+
+BEGIN;
+UPDATE animals SET species='unspecified';
+SELECT * FROM ANIMALS ;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals SET species='digimon'
+WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon'
+WHERE species is NULL;
+SELECT * FROM animals;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+SAVEPOINT delete1;
+DELETE FROM animals WHERE date_of_birth > 'jan-1-2022';
+UPDATE animals SET weight_kg = weight_kg*-1;
+ROLLBACK TO delete1;
+UPDATE animals SET weight_kg = weight_kg*-1
+WHERE weight_kg <0;
+SELECT * FROM animals; 
+COMMIT;
+
+SELECT COUNT(*) as total_number_of_animals FROM animals;
+SELECT COUNT(*) as Never_escaped from animals WHERE escape_attempts=0;
+SELECT AVG(weight_kg) as average_weight FROM animals;
+SELECT neutered,MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species,MIN(weight_kg) as minimum_weight, MAX(weight_kg) as maximum_weight FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) as average_escape_attempts FROM animals WHERE date_of_birth BETWEEN 'jan-1-1990' AND 'jan-1-2000'
+GROUP BY species;
```

Comment:
- [ ] This query is to know the average number of escape attempts per animal type of those born between 1990 and 2000 which means all the year 2000 `Jan to Dec`, the dates should be `'jan-1-1990' AND 'dec-31-2000'`. Kindly fix that 🙃